To: vim_dev@googlegroups.com
Subject: Patch 7.3.233
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.233
Problem:    ":scriptnames" and ":breaklist" show long file names.
Solution:   Shorten to use "~/" when possible. (Jean-Rene David)
Files:      src/ex_cmds2.c 


*** ../vim-7.3.232/src/ex_cmds2.c	2011-05-19 18:26:34.000000000 +0200
--- src/ex_cmds2.c	2011-06-26 04:20:24.000000000 +0200
***************
*** 682,691 ****
  	for (i = 0; i < dbg_breakp.ga_len; ++i)
  	{
  	    bp = &BREAKP(i);
  	    smsg((char_u *)_("%3d  %s %s  line %ld"),
  		    bp->dbg_nr,
  		    bp->dbg_type == DBG_FUNC ? "func" : "file",
! 		    bp->dbg_name,
  		    (long)bp->dbg_lnum);
  	}
  }
--- 682,693 ----
  	for (i = 0; i < dbg_breakp.ga_len; ++i)
  	{
  	    bp = &BREAKP(i);
+ 	    if (bp->dbg_type == DBG_FILE)
+ 		home_replace(NULL, bp->dbg_name, NameBuff, MAXPATHL, TRUE);
  	    smsg((char_u *)_("%3d  %s %s  line %ld"),
  		    bp->dbg_nr,
  		    bp->dbg_type == DBG_FUNC ? "func" : "file",
! 		    bp->dbg_type == DBG_FUNC ? bp->dbg_name : NameBuff,
  		    (long)bp->dbg_lnum);
  	}
  }
***************
*** 3268,3274 ****
  
      for (i = 1; i <= script_items.ga_len && !got_int; ++i)
  	if (SCRIPT_ITEM(i).sn_name != NULL)
! 	    smsg((char_u *)"%3d: %s", i, SCRIPT_ITEM(i).sn_name);
  }
  
  # if defined(BACKSLASH_IN_FILENAME) || defined(PROTO)
--- 3270,3280 ----
  
      for (i = 1; i <= script_items.ga_len && !got_int; ++i)
  	if (SCRIPT_ITEM(i).sn_name != NULL)
! 	{
! 	    home_replace(NULL, SCRIPT_ITEM(i).sn_name,
! 						    NameBuff, MAXPATHL, TRUE);
! 	    smsg((char_u *)"%3d: %s", i, NameBuff);
!         }
  }
  
  # if defined(BACKSLASH_IN_FILENAME) || defined(PROTO)
*** ../vim-7.3.232/src/version.c	2011-06-26 04:01:37.000000000 +0200
--- src/version.c	2011-06-26 04:23:48.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     233,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
223. You set up a web-cam as your home's security system.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
